src/components/Grid/Row.js

```diff
@@ -107,32 +110,50 @@ const getFullWidthStyles = props => {
     justifyContent: 'center',
     [responsive.xs]: {
       display: 'block',
-      margin: `0 calc(-1 * (100vw - ${screenWidths.xs}px) / 2)`
+      marginTop: `0 calc(-1 * (100vw - ${screenWidths.xs}px) / 2)`,
+      marginRight: `0 calc(-1 * (100vw - ${screenWidths.xs}px) / 2)`,
+      marginBottom: `0 calc(-1 * (100vw - ${screenWidths.xs}px) / 2)`,
+      marginLeft: `0 calc(-1 * (100vw - ${screenWidths.xs}px) / 2)`,
     },
 
     [responsive.sm]: {
       display: 'block',
-      margin: `0 calc(-1 * (100vw - ${screenWidths.sm}px) / 2)`
+      marginTop: `0 calc(-1 * (100vw - ${screenWidths.sm}px) / 2)`,
+      marginRight: `0 calc(-1 * (100vw - ${screenWidths.sm}px) / 2)`,
+      marginBottom: `0 calc(-1 * (100vw - ${screenWidths.sm}px) / 2)`,
+      marginLeft: `0 calc(-1 * (100vw - ${screenWidths.sm}px) / 2)`,
     },
 
     [responsive.md]: {
       display: 'block',
-      margin: `0 calc(-1 * (100vw - ${screenWidths.md}px) / 2)`
+      marginTop: `0 calc(-1 * (100vw - ${screenWidths.md}px) / 2)`,
+      marginRight: `0 calc(-1 * (100vw - ${screenWidths.md}px) / 2)`,
+      marginBottom: `0 calc(-1 * (100vw - ${screenWidths.md}px) / 2)`,
+      marginLeft: `0 calc(-1 * (100vw - ${screenWidths.md}px) / 2)`,
     },
 
     [responsive.mdLg]: {
       display: 'block',
-      margin: `0 calc(-1 * (100vw - ${screenWidths.mdLg}px) / 2)`
+      marginTop: `0 calc(-1 * (100vw - ${screenWidths.mdLg}px) / 2)`,
+      marginRight: `0 calc(-1 * (100vw - ${screenWidths.mdLg}px) / 2)`,
+      marginBottom: `0 calc(-1 * (100vw - ${screenWidths.mdLg}px) / 2)`,
+      marginLeft: `0 calc(-1 * (100vw - ${screenWidths.mdLg}px) / 2)`,
     },
 
     [responsive.lg]: {
       display: 'block',
-      margin: `0 calc(-1 * (100vw - ${screenWidths.lg}px) / 2)`
+      marginTop: `0 calc(-1 * (100vw - ${screenWidths.lg}px) / 2)`,
+      marginRight: `0 calc(-1 * (100vw - ${screenWidths.lg}px) / 2)`,
+      marginBottom: `0 calc(-1 * (100vw - ${screenWidths.lg}px) / 2)`,
+      marginLeft: `0 calc(-1 * (100vw - ${screenWidths.lg}px) / 2)`,
     },
 
     [responsive.xl]: {
       display: 'block',
-      margin: `0 calc(-1 * (100vw - ${screenWidths.xl}px) / 2)`
+      marginTop: `0 calc(-1 * (100vw - ${screenWidths.xl}px) / 2)`,
+      marginRight: `0 calc(-1 * (100vw - ${screenWidths.xl}px) / 2)`,
+      marginBottom: `0 calc(-1 * (100vw - ${screenWidths.xl}px) / 2)`,
+      marginLeft: `0 calc(-1 * (100vw - ${screenWidths.xl}px) / 2)`,
```

Comment:
Will do

Reply:
I refactored it a bit.  Also improved performance since now the styles are built once instead of each render.  There's still a bit of duplication, but I think it's down to a reasonable level now.